packages/lit-dev-content/site/tutorials/content/svg-templates/04.md

```diff
@@ -0,0 +1,113 @@
+This section of the tutorial covers the basics of the `<pattern>`
+element in SVG including how to:
+
+- Define a `<pattern>` element
+- Apply a `<pattern>` as a fill
+
+### Learn
+
+In SVG, the `<pattern>` element visually repeats children across
+a 2D plane.
+
+The [`patternUnits`](https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/patternUnits)
+property defines the relative coordinate geometry used to paint a
+pattern. Setting `patternUnits` to `userSpaceOnUse` makes
+geometry relative to the user space. In this case, user space is the
+DOM.
+
+```ts
+const helloPattern = svg`
+  <pattern patternUnits="userSpaceOnUse">
+    ${createTile()}
+  </pattern>
+`;
+```
+
+The `<pattern>` element can be given an `id` and referenced as a
+[`fill`](https://developer.mozilla.org/en-US/docs/Web/SVG/Tutorial/Fills_and_Strokes)
+in other SVG elements as shown in the example below:
+
+```ts
+const helloPattern = svg`
+  <pattern
+    id="hello-pattern"
+    patternUnits="userSpaceOnUse">
+    ${createTile()}
+  </pattern>
+`;
+
+const helloPatternFill = svg`
+  <rect fill="url(#hello-pattern)" width="200" height="200"></rect>
+`;
+```
+
+### Apply
+
+Create a function called `createRepeatPattern` that returns a pattern
+element. Call `createTile` inside a `<pattern>` element. Give the
+`<pattern>` an `id` of `repeat-pattern`.
+
+```ts
+const createRepeatPattern = () => svg`
+  <pattern id="repeat-pattern">
+    ${createTile()}
+  </pattern>
+`;
+```
+
+Provide [pattern attributes](https://developer.mozilla.org/en-US/docs/Web/SVG/Element/pattern#attributes)
+like `height` and `width` to define how a tiles are painted. The example
+below creates a 200px square tile with a vertical and horizontal offset of
+`-10`. The pattern is painted using geometry defined by `userSpaceOnUse`.
+
+```ts
+const createRepeatPattern = () => svg`
+  <pattern
+    id="repeat-pattern"
+    x="-10"
+    y="-10"
+    width="200"
+    height="200"
+    patternUnits="userSpaceOnUse">
+    ${createTile()}
+  </pattern>
+`;
+```
+
+Place `createRepeatPattern` inside the `<defs>` element in `render`
+to make `#repeat-pattern` available to other elements.
+
+```ts
+render() {
+  return html`
+    <svg width="100%" height="100%">
+      <defs>
+        ...
+        ${createRepeatPattern()}
+      </defs>
+    </svg>
+  `;
+}
+```
+
+Add two `<rect>` elements to the `render` method in  `repeat-pattern`. One
```

Comment:
Need to tell user to remove  `${createTile()}`

Reply:
done!